### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Kitty-Sam**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,51 +1,72 @@
...
+    },
+    "dependencies": {
+        "@react-native-async-storage/async-storage": "^1.17.11",
+        "@react-navigation/material-top-tabs": "^6.5.1",
+        "@react-navigation/native": "^6.1.1",
+        "@react-navigation/native-stack": "^6.9.7",
+        "@react-navigation/stack": "^6.3.10",
+        "react": "18.1.0",
+        "react-native": "0.70.6",
+        "react-native-pager-view": "^6.1.2",
```
Comment (stanislavlebedyantsev) [inline package.json:28]: unused dependency

File: src/components/Column/index.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import React, { FC } from 'react';
+import { View } from 'react-native';
+import { styles } from '~components/Column/style';
+import { ColumnPropsType } from '~components/Column/type';
+
+export const Column: FC<ColumnPropsType> = ({ children }) => {
```
Comment (stanislavlebedyantsev) [inline src/components/Column/index.tsx]: you don't need define interface for component which just receive children
we have build in type called PropsWithChild

File: src/components/Display/index.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export const Display: FC<DisplayPropsType> = ({ input, result, isEqual }) => {
+    const { colors } = useTheme();
+    return (
+        <ErrorBoundary>
+            <View style={styles.container}>
+                {!isEqual && isEqual !== undefined ? (
+                    <Text style={[styles.text, { color: colors.border }]}>{input}</Text>
+                ) : (
+                    <></>
```
Comment (stanislavlebedyantsev) [inline src/components/Display/index.tsx]: you should not use ternary operator here

File: src/components/Display/style.ts [hunk 4]
```diff
@@ -1,13 +1,17 @@
...
-    alignItems: 'center',
-    justifyContent: 'center',
-  },
+    container: {
+        height: '39%',
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    text: {
+        fontSize: 32,
```
Comment (stanislavlebedyantsev) [inline src/components/Display/style.ts:18]: why this one not using theme object?

File: src/components/DisplayClass/index.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+export default class DisplayClass extends React.Component<DisplayClassPropsType, DisplayClassStateType> {
+    render() {
+        const { input, result, isEqual } = this.props;
+        return (
+            <ErrorBoundary>
+                <View style={styles.container}>
+                    {!isEqual && isEqual !== undefined ? (
+                        <Text style={[styles.text, { color: DarkTheme.colors.primary }]}>{input}</Text>
+                    ) : (
+                        <></>
```
Comment (stanislavlebedyantsev) [inline src/components/DisplayClass/index.tsx]: not needed

File: src/components/DisplayClass/style.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+};
+
+export const styles = StyleSheet.create<DisplayStyleType>({
+    container: {
+        height: '39%',
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    text: {
+        fontSize: 32,
```
Comment (stanislavlebedyantsev) [inline src/components/DisplayClass/style.ts:15]: same for theme

File: src/components/Header/Header.tsx [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import React, { FC } from 'react';
+import { Text } from 'react-native';
+import { styles } from '~components/Header/style';
+import { useTheme } from '@react-navigation/native';
+import { HeaderPropsType } from '~components/Header/type';
+
+export const Header: FC<HeaderPropsType> = ({ header }) => {
```
Comment (stanislavlebedyantsev) [inline src/components/Header/Header.tsx]: it's better to call this props as title, not header

File: src/components/KeyPad/index.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+import { styles } from './style';
+import { Button } from '~components/Button';
+import { KeyPadType } from '~components/KeyPad/type';
+
+export const KeyPad: FC<KeyPadType> = memo(({ onPressHandler }) => {
+    return (
+        <View style={styles.rootContainer}>
+            <Row>
+                <View style={styles.leftButtonsContainer}>
+                    {leftButtonsLabels.map(({ title, shape, type }) => (
```
Comment (stanislavlebedyantsev) [inline src/components/KeyPad/index.tsx]: Why you don't use flatlist here?

File: src/components/KeyPad/index.tsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import React, { FC, memo } from 'react';
+import { Row } from '~components/Row';
+import { Column } from '~components/Column';
+import { FlatList, View } from 'react-native';
+import { leftButtonsLabels, rightButtonsLabels } from '~constants/buttonsLabels/buttonsLabels';
+import { styles } from './style';
+import { Button } from '~components/Button';
+import { KeyPadType } from '~components/KeyPad/type';
+
+export const KeyPad: FC<KeyPadType> = memo(({ onPressHandler }) => {
```
Comment (stanislavlebedyantsev) [inline src/components/KeyPad/index.tsx:14]: Usually it's better to provide callback with checks as 2nd parameter of memo, but for now it's okay

File: src/components/KeyPad/index.tsx [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+                            shape={shape}
+                            onPress={() => onPressHandler(title)}
+                            type={type}
+                        />
+                    ))}
+                </View>
+                <Column>
+                    <FlatList
+                        data={rightButtonsLabels}
+                        renderItem={({ item: { shape, title, type } }) => (
```
Comment (stanislavlebedyantsev) [inline src/components/KeyPad/index.tsx]: it's better to contain render function as separate and wrap it in usecallback

File: src/components/KeyPad/index.tsx [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+                            <Button onPress={() => onPressHandler(title)} title={title} shape={shape} type={type} />
```
Comment (stanislavlebedyantsev) [inline src/components/KeyPad/index.tsx]: let's avoid inline callbacks

File: src/components/KeyPadClass/index.tsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+import { Column } from '~components/Column';
+import { FlatList, View } from 'react-native';
+import { leftButtonsLabels, rightButtonsLabels } from '~constants/buttonsLabels/buttonsLabels';
+import { styles } from './style';
+import { Button } from '~components/Button';
+
+export interface KeyPadClassPropsType {
+    onPressHandler: (text: string) => void;
+}
+export interface KeyPadClassStateType {}
```
Comment (stanislavlebedyantsev) [inline src/components/KeyPadClass/index.tsx]: is this interface really needed here?

File: src/components/KeyPadClass/index.tsx [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+
+export default class KeyPadClass extends PureComponent<KeyPadClassPropsType, KeyPadClassStateType> {
+    render() {
+        const { onPressHandler } = this.props;
+        return (
+            <View style={styles.rootContainer}>
+                <Row>
+                    <View style={styles.leftButtonsContainer}>
+                        {leftButtonsLabels.map(({ title, shape, type }) => (
```
Comment (stanislavlebedyantsev) [inline src/components/KeyPadClass/index.tsx]: flatlist

File: src/components/Row/type.ts [hunk 14]
```diff
@@ -0,0 +1,5 @@
+import { ReactNode } from 'react';
+
+export interface RowPropsType {
```
Comment (stanislavlebedyantsev) [inline src/components/Row/type.ts:3]: not needed, because we have build in type in react

File: src/screens/CalculatorCCScreen/style.ts [hunk 15]
```diff
@@ -1,11 +1,11 @@
...
-  rootContainer: ViewStyle;
+    rootContainer: ViewStyle;
 };
 
 export const styles = StyleSheet.create<CalculatorScreenStyleType>({
-  rootContainer: {
-    marginHorizontal: 48,
-  },
+    rootContainer: {
+        marginHorizontal: 24,
```
Comment (stanislavlebedyantsev) [inline src/screens/CalculatorCCScreen/style.ts:9]: also can be moved to theme

**Review by stanislavlebedyantsev — CHANGES_REQUESTED** [review]

File: src/components/Column/index.tsx (hunk 2 above)
Comment (Kitty-Sam) [reply src/components/Column/index.tsx]: nice solution

File: src/components/KeyPad/index.tsx (hunk 8 above)
Comment (Kitty-Sam) [reply src/components/KeyPad/index.tsx]: Because Flatlist doesnt support flexWrap: wrap

**Pull request merged**